### DISCUSSION TITLE
Do-while loop?

### DISCUSSION THREAD

**Discussion by felipecrs** [twpayne/chezmoi#1671] [Q&A]
I'm trying to do a simple `do` `while` loop, until a condition is met. I searched everywhere and I could not find a way to make it work.

For the `go/template` experts, do you have any tip for me?

This is what I'm trying to do, by the way:

```
{{ $name := "" -}}
{{- if hasKey . "name" -}}
{{-   $name = .name -}}
{{- end -}}
{{- if and stdinIsATTY (not $chezmoiForce) -}}
{{-   $question := "What is your full name" -}}
{{-   do -}}
{{-     if $name -}}
{{-       $name = promptString $question $name -}}
{{-     else -}}
{{-       $name = promptString $question -}}
{{-     end -}}
{{-   while $name -}}
{{-   end -}}
{{-     writeToStdout (printf "✔ Name set as '%s'\n" $name) -}}        
{{- end -}}
```

**Comment by twpayne** [comment]
I don't think this is possible. `text/template` is a templating language designed to render static data as text and consequently has very few control structures.

**Reply by felipecrs** [reply]
Yeah, but regardless I need to get the job done. [It seems that in Go 1.18 there will be](https://github.com/golang/go/issues/20531) `continue` and `break` keywords for using inside of a `range`, so that I can do something like:

```
{{ range := until 99 }}
ask for name
{{ if name is valid }}
break
{{ end }} 
{{ end }}
```

My intention with this question was to find possible workarounds, as I'm already aware of the limitations of `text/template`.

**Reply by twpayne** [reply]
Of course as soon as Go 1.18 comes out the "official" releases of chezmoi will be built with it. In the short time, I'm not aware of any workarounds, but am of course keen to learn about any.

**Reply by felipecrs** [reply]
Yep. Thank you!
